### PR TITLE
Add telemetry for OmniSharp server requests

### PR DESCRIPTION
**Pull request by DustinCampbell**
This change adds telemetry tracking for the time elapsed from the point that a request sent to the OmniSharp server to the point that a response is received. To keep telemetry traffic low, we simply track the counts of various ranges of delay times for each type of request.

In addition, I refactored the telemetry a bit in coreclr-debug.ts to ensure that we can share the same TelemetryReporter.

cc @gregg-miskelly, @chuckries, @Pilchie

### CODE REVIEW

**Comment by gregg-miskelly** [comment]
:+1:

**Comment by gregg-miskelly** [comment]
Sorry, that comment was supposed to be for the other PR. I haven't had a chance to look at this one yet. Give me a few minutes...

File: package.json [hunk 1]
```diff
@@ -157,7 +157,6 @@
         "windows": {
           "program": "./coreclr-debug/debugAdapters/OpenDebugAD7.exe"
         },
-        "aiKey": "AIF-d9b70cd4-b9f9-4d70-929b-a071c400b217",
```
Comment (gregg-miskelly) [inline package.json]: We need this. VS Code reads this value.

Comment (DustinCampbell) [inline package.json]: It does? The instructions for VS Code telemetry do not say to add this to the debuggers section. Or, is this used for the debug adapter telemetry (which I'm totally unfamiliar with).

File: src/omnisharpServer.ts [hunk 2]
```diff
@@ -57,9 +135,30 @@ export abstract class OmnisharpServer {
 	private _setState(value: ServerState) : void {
 		if (typeof value !== 'undefined' && value !== this._state) {
 			this._state = value;
-			this._fireEvent('stateChanged', this._state);
+			this._fireEvent(Events.StateChanged, this._state);
 		}
 	}
+    
+    private _recordDelay(path: string, elapsedTime: number) {
```
Comment (gregg-miskelly) [inline src/omnisharpServer.ts]: Might be obvious to other people, but I am having trouble figuring out what 'path' means.

File: src/omnisharpServer.ts [hunk 3]
```diff
@@ -57,9 +135,30 @@ export abstract class OmnisharpServer {
...
+            delays = new Delays();
+            this._telemetryDelays[path] = delays;
+        }
+        
+        delays.report(elapsedTime);
+    }
+    
+    public reportAndClearTelemetry() {
+        for (var path in this._telemetryDelays) {
+            const eventName = 'omnisharp' + path;
```
Comment (gregg-miskelly) [inline src/omnisharpServer.ts]: If 'path' is a file path, might that have PII?

Comment (DustinCampbell) [inline src/omnisharpServer.ts]: no it doesn't. It's the "path" to the request. You're right. I'll pick a better name here.

File: package.json (hunk 1 above)
Comment (gregg-miskelly) [inline package.json]: Yes, it is used with debug adapter telemetry.

**Comment by gregg-miskelly** [comment]
:+1:

File: package.json (hunk 1 above)
Comment (DustinCampbell) [inline package.json]: OK -- will add that back. Sorry about that! I wasn't sure because the debugger code just dug it out of package.json.

**Comment by DustinCampbell** [comment]
@gregg-miskelly I implemented your feedback. Look better now?

**Comment by gregg-miskelly** [comment]
:+1:

**Comment by Pilchie** [comment]
:+1:

**Pull request merged**